### PR TITLE
Add application 'builds' for CI jobs and build results

### PR DESCRIPTION
**Pull request by txels**
Following on the previous PR's footsteps, these are additional models for a `builds` app.

I think it's important that we get the data model right to begin with, then we can start building the providers and the front-end based on that.

Next step, a `features` app.

### CODE REVIEW

File: ployst/builds/models.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    UNSTABLE = Status('yellow', 'unstable')
+    FAILURE = Status('red', 'failure')
+    NOT_BUILT = Status('gray', 'not built')
+    ALL_STATUSES = [NOT_BUILT, FAILURE, UNSTABLE, SUCCESS]
+
+    job = models.ForeignKey(Job, related_name='builds')
+    revision = Revision()
+    status = models.CharField(
+        max_length=6, default=NOT_BUILT.colour, choices=ALL_STATUSES
+    )
```
Comment (txels) [inline ployst/builds/models.py]: I'm not yet sure how much data we want to store for each build and how. E.g. test results etc.
Also I'd like a second opinion about whether we actually need builds and jobs or just builds.

Comment (alexcouper) [inline ployst/builds/models.py]: I'm not sure what the use-case is for Jobs.

Do you have something in mind?

Comment (txels) [inline ployst/builds/models.py]: I was literally just turning this diagram into models https://docs.google.com/drawings/d/1tWhxPzuBWYWlxoo4mueNnkUcSG3HrK6691bS6XbQ_xw/edit and found this point and wanted to trigger a discussion. I can get rid of it and add the job name/URL to the build.

Comment (txels) [inline ployst/builds/models.py]: Added a `job_id` that should make sense to the CI provider. Then it's up to the provider implementation to store sthg that will make it able to access the job and trigger a build etc.

File: ployst/builds/models.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    Status = namedtuple('BuildStatus', ['colour', 'result'])
+    Status.__str__ = lambda self: self.result
+
+    SUCCESS = Status('green', 'success')
+    UNSTABLE = Status('yellow', 'unstable')
+    FAILURE = Status('red', 'failure')
+    NOT_BUILT = Status('gray', 'not built')
+    ALL_STATUSES = [NOT_BUILT, FAILURE, UNSTABLE, SUCCESS]
+
+    provider = models.CharField(
```
Comment (txels) [inline ployst/builds/models.py:22]: Doubting whether we want a model for providers and a foreign key here.
But as providers will need to be added to the codebase, I don't think it really makes sense to manage them in the database.

**Comment by txels** [comment]
I got rid of the Job model, not really needed here. Added data in the build that the provider can use to retrieve job information etc.

File: ployst/builds/models.py (hunk 2 above)
Comment (alexcouper) [inline ployst/builds/models.py:22]: Yeah i think this makes sense for now - we can always change in future.

File: ployst/builds/migrations/0001_initial.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
```
Comment (alexcouper) [inline ployst/builds/migrations/0001_initial.py:1]: Currently working on a project that doesn't have the luxury of south.

Glad to have it back!

**Pull request merged**